velox/functions/prestosql/aggregates/tests/AverageAggregationTest.cpp

```diff
@@ -199,5 +203,76 @@ TEST_F(AverageAggregationTest, partialResults) {
 
   assertQuery(plan, "SELECT row(4950, 100)");
 }
+
+TEST_F(AverageAggregationTest, avgDecimal) {
+  // facebook::velox::exec::test::AggregateTypeResolver resolver(step);
+  auto runAndCompare = [&](const std::string& exprStr,
+                           std::vector<VectorPtr> input,
+                           VectorPtr expectedResult,
+                           bool isSingle = true) {
+    // Decimal average cannot be directly compared with duckdb decimal
+    // average whose return type is double. Hence, cannot use testAggregations.
+    // Need to use PlanBuilder as it registers a AggregateTypeResolver Vs
+    // evaluate<ReturnType> which would compile the expressions through
+    // Simple/Vector function resolvers.
+    PlanBuilder builder;
+    builder.values({makeRowVector(input)});
+    if (isSingle) {
+      builder.singleAggregation({}, {"avg(c0)"});
+    } else {
+      builder.partialAggregation({}, {"avg(c0)"});
```

Comment:
You should do the full "partial + local exchange + intermediate + local exchange + final"

Reply:
Agree, I need to still implement the intermediate aggregation.